src/__/collections/groupBy.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace collections;
+
+/**
+ * Returns an associative array where the keys are values of $key.
+ *
+ * Based on {@author Chauncey McAskill}'s {@link https://gist.github.com/mcaskill/baaee44487653e1afc0d array_group_by()}
+ * function.
+ *
+ * @param array $array
+ * @param int|float|string|\Closure $key
+ *
+ * @return array
+ *
+ */
+function groupBy(array $array, $key)
+{
+    if (!\is_bool($key) && !\is_scalar($key) && !\is_callable($key)) {
+        return $array;
```

Comment:
I wonder if we should throw an exception in this case.

Reading the rest of the lib, just returning the value seems what have be done so far
https://github.com/maciejczyzewski/bottomline/blob/0228b5a8bf9bcb267c8b0d1833db833cb162861a/src/__/collections/set.php#L22

Reply:
Yea, I think an exception would be helpful to let users know why an array wasn't grouped; just wasn't sure about staying consistent. Should I go with an exception then?

Reply:
I will opt for the more explicit behavior: exception.

@maciejczyzewski any thought on that?

Reply:
Such error messages would be helpful for users working with our library, they must be concrete and concise.

@MonsieurV Probably, it should be a separate module that will handle errors/exceptions. Because many messages will be repeated/or similar. In code it should be just a call to appropriate ID of warning/error.

Reply:
Good idea, that will ensure consistency.